src/main/java/mate/academy/spring/service/impl/RoleServiceImpl.java

```diff
@@ -0,0 +1,32 @@
+package mate.academy.spring.service.impl;
+
+import java.util.Arrays;
+import mate.academy.spring.dao.RoleDao;
+import mate.academy.spring.model.Role;
+import mate.academy.spring.service.RoleService;
+import org.springframework.stereotype.Service;
+
+@Service
+public class RoleServiceImpl implements RoleService {
+    private final RoleDao roleDao;
+
+    public RoleServiceImpl(RoleDao roleDao) {
+        this.roleDao = roleDao;
+    }
+
+    @Override
+    public Role add(Role role) {
+        return roleDao.add(role);
+    }
+
+    @Override
+    public Role getByName(String roleName) {
+        Role.RoleName roleEnum = Arrays.stream(Role.RoleName.values())
+                .filter(r -> r.name().equalsIgnoreCase(roleName))
+                .findFirst()
+                .orElseThrow(() ->
+                        new RuntimeException("Role with name " + roleName + " doesn't exist"));
```

Comment:
Why do you need this code? You can just get a role from DB

Reply:
Because much faster receive role from code than from DB. Isn't it?